### DISCUSSION TITLE
python client api?

### DISCUSSION THREAD

**Discussion by SiNa88** [k3s-io/k3s#7769] [Q&A] [answered]
Hi all,

I was wondering whether K3s supports the Python client API such as the following?
https://github.com/kubernetes-client/python

Thanks in advance

**Comment by htmslh** [comment]
I tried to run the example script inside a k3s pod and it couldn't reach the kubernetes API.
The incluster config loads fine but whenever I try to access any of the API routes, it timed out trying to reach the cluster's IP.

Running the same python script in minikube pod and it works out of the box.

**Reply by brandond** [reply, marked as answer]
https://docs.k3s.io/quick-start#install-script
> A [kubeconfig](https://kubernetes.io/docs/concepts/configuration/organize-cluster-access-kubeconfig/) file will be written to `/etc/rancher/k3s/k3s.yaml` and the kubectl installed by K3s will automatically use it

If you're using other clients, you probably need to `export KUBECONFIG=/etc/rancher/k3s/k3s.yaml`, or pass the path via the appropriate CLI flag.

This is just basic Kubernetes client stuff.

**Reply by htmslh** [reply]
Thanks for the response! I did make a symlink at the default `~/.kube/config` location, because this was needed if I want to use the python client in general, locally, not inside a pod or a container.

The problem is when I deploy it into the cluster. The error log shows that it tries to make a request to the cluster IP shown in `kubectl get services` but it timed out. I've set up a service account & role for this too but it still doesn't work.
